src/Features/Core/Portable/CodeFixes/Suppression/ConfigureSeverityLevelCodeAction.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Suppression
+{
+    internal sealed class ConfigureSeverityLevelCodeAction : CodeAction.CodeActionWithNestedActions
+    {
+        public ConfigureSeverityLevelCodeAction(
+            Diagnostic diagnostic,
+            ImmutableArray<CodeAction> nestedActions)
+            : base("Configure severity level via editorconfig file",
+            ImmutableArray<CodeAction>.CastUp(nestedActions), isInlinable: false)
+        {
+        }
+
+        internal static Task<Solution> ConfigureEditorConfig(
+            string severity,
+            Diagnostic diagnostic,
+            Project project,
+            Dictionary<string, Option<CodeStyleOption<bool>>> languageSpecificOptions,
+            Dictionary<string, Option<CodeStyleOption<ExpressionBodyPreference>>> expressionOptions,
+            string language,
+            CancellationToken cancellationToken)
+        {
+            var solution = project.Solution;
+            TextDocument editorconfig = null;
+            FindOrGenerateEditorconfig();
+
+            editorconfig.TryGetText(out var text);
+            var result = editorconfig.GetTextAsync().Result;
```

Comment:
Can you instead make this an async method `internal static async Task<Solution> ConfigureEditorConfigAsync(...)` and use `var result = await editorconfig.GetTextAsync().ConfigureAwait(false)` here?